airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -47,9 +47,13 @@ def check_connection(self, source: Source, logger: logging.Logger, config: Mappi
         return True, None
 
     def _get_stream_slice(self, stream):
-        slices = stream.stream_slices(
-            cursor_field=stream.cursor_field,
-            sync_mode=SyncMode.full_refresh,
+        # We wrap the return output of stream_slices() because some implementations return types that are iterable,
+        # but not iterators such as lists or tuples
+        slices = iter(
```

Comment:
given that some output of `stream_slices()` could return something that is already an iterator, but calling `iter()` on an object that is already an iterator ends up with the same result. I experimented with this and added a test case for this.

The alternative is we use isinstance to check for the iterator type, but it seems like unnecessary clutter to the code that would yield the same result

Reply:
<3 for adding a testcase for calling `iter` on an iterator